### ISSUE TITLE
How to make Sonoff THs internal hardware button/switch functional again?

### ISSUE THREAD

**Issue by lev2k** [colinl/Sonoff-Tasmota#7]
When I don't need the PID Control, (which, by the way, works absolutely great for me), I would like to use Sonoffs internal hardware button/switch again.
I know, I can turn on/off the relay via MQTT commands, but when using the button, the relay switches off again, immediately after pressing it.
I've also tried a rule like: "rule on button1#state do publish cmnd/sonoff-TH16/PID_MANUAL_POWER toggle endon"
But "toggle" only works, if the relay is in "on" state.
Any ideas?

**Comment by colinl** [comment]
I don't think you will be able to do this inside the THnn as the timeprop s/w will repeatedly force the relay to the state it wants.  However, if you have node-red or openhab or something similar running somewhere then you could configure the TH to send the button state via MQTT and when you see that, put the pid into manual and adjust the manual power to either 1 or 0 to switch the relay on or off.

**Comment by colinl** [comment]
You could also use this technique to make a long press put it into manual mode, then use quick press to change the state, and long press again to put it back into auto.
